frontend/app/student/run-simulation/[instanceId]/page.tsx

```
@@ -1112,104 +1760,128 @@ ${availablePersonas.map(persona => `• @${persona.name.toLowerCase().replace(/\
+                          <div className="flex flex-col gap-2">
+                            <div className="text-sm text-gray-600 font-medium">{(message as any).sceneLoading ? 'Loading next scene...' : (message as any).gradingInProgress ? 'Submitting for grading...' : 'Processing your message...'}</div>
+                            <div className="w-full h-1.5 bg-gray-200/60 rounded-full overflow-hidden backdrop-blur-sm">
+                              <div className="h-full bg-gradient-to-r from-blue-400 via-blue-500 to-blue-600 animate-gradient rounded-full" style={{ width: '100%', backgroundSize: '200% 100%' }}></div>
+                            </div>
                           </div>
+                        ) : (
+                          message.text.split('\n').map((line, index) => {
+                            const boldFormatted = line.replace(/\*\*(.*?)\*\*/g, '<strong class="font-semibold">$1</strong>')
+                            return (
+                              <div key={index} dangerouslySetInnerHTML={{ __html: boldFormatted }} />
+                            )
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)